frontend/src/App.vue

```diff
@@ -1,112 +1,88 @@
-}
+const { windowWidth, windowHeight, offsetX, offsetY, viewBoxAttr } = viewBox
 
-const _boardToScreenCoordinates = (pos: { x: number; y: number }) => {
-  return { x: pos.x - offsetX.value, y: pos.y - offsetY.value }
+function onMessage(messageEvent: MessageEvent) {
+  const data: unknown = JSON.parse(messageEvent.data)
+  if (isDrawCircleEvent(data)) {
+    const circle: IBoardElementCircle = data.shape;
+    boardElements.push(circle)
+  } else {
+    console.log("data is not a circle.")
+  }
 }
 
-const isDragging = ref(false)
-
-const onMouseDown = (event: MouseEvent) => {
-  switch (event.button) {
-    case 0: {
-      // left click
-      const center = screenToBoardCoordinates({
-        x: event.clientX,
-        y: event.clientY,
-      })
-      boardElements.push({
-        id: nanoid(),
-        kind: 'circle',
-        cx: center.x,
-        cy: center.y,
-        radius: 5,
-      })
-      break
+function drawCircle(event: MouseEvent) {
+  const center = screenToBoardCoordinates(
+    {
+      x: event.clientX,
+      y: event.clientY,
+    },
+    {
+      x: offsetX.value,
+      y: offsetY.value,
+    },
+  )
+  const circle: IBoardElementCircle = {
+    id: nanoid(),
+    kind: 'circle',
+    cx: center.x,
+    cy: center.y,
+    radius: 5,
+  }
+  boardElements.push(circle)
+  if (isOpenWebSocket(board.socket)) {
+    const boardEvent: IDrawCircleEvent = {
+      eventType: 'draw-circle',
+      shape: circle,
     }
-    case 1: // middle/wheel click
-    case 2: // right click
-      isDragging.value = true
-      break
-    default:
-      break
+    board.socket.send(JSON.stringify(boardEvent))
   }
 }
 
-const onMouseUp = () => {
-  isDragging.value = false
+// Viewbox dragging behavior
+const isDragging = ref(false);
+const startDrag = () => {
+  isDragging.value = true;
 }
-
-const onMouseMove = (event: MouseEvent) => {
-  mouseScreenX.value = event.clientX
-  mouseScreenY.value = event.clientY
-
+const dragViewbox = (event: MouseEvent) => {
   if (isDragging.value) {
     offsetX.value -= event.movementX
     offsetY.value -= event.movementY
   }
 }
+const endDrag = () => {
+  isDragging.value = false
+}
 
-onMounted(() => {
-  window.addEventListener('resize', onWindowResized)
-  window.addEventListener('mouseup', onMouseUp)
-  window.addEventListener('mousedown', onMouseDown)
-  window.addEventListener('mousemove', onMouseMove)
-})
-
-onUnmounted(() => {
-  window.removeEventListener('resize', onWindowResized)
-  window.removeEventListener('mouseup', onMouseUp)
-  window.removeEventListener('mousedown', onMouseDown)
-  window.removeEventListener('mousemove', onMouseMove)
-})
 </script>
 
 <template>
-  <svg
-    class="cursor"
-    :style="cursorStyle"
-    viewBox="0 0 32 32"
-    width="32"
-    height="32"
-  >
-    <path d="M0,0 L32,8 L16,16 L8,32 Z" fill="black"></path>
-  </svg>
+  <BoardCursor
+    @leftClick="drawCircle"
+    @dragStart="startDrag"
+    @move="dragViewbox"
+    @dragEnd="endDrag"
+  />
```

Comment:
I just did this as a sort of experiment - I think its probably for the best if we just use the built-in cursors. I believe these are even customizable.